text/0000-match-ergonomics.md

```diff
@@ -3,11 +3,15 @@
 - RFC PR: (leave this empty)
 - Rust Issue: (leave this empty)
 
+I'm writing comments inline (not making changes), hopefully they'll stand out in the diff.
+
 # Summary
 [summary]: #summary
 
 Better ergonomics for pattern-matching on references.
 
+AIUI, this will only apply to match and similar expressions, not pattern matching in general?
```

Comment:
hmm, we should probably dig into this a bit - how does this look when calling a function? Does it introduce auto-ref in places it didn't exist before? Is it still backwards compatible? It is also worth thinking about (and giving some examples in the RFC) of how this will work with match arms with multiple patterns and/or guards - I don't expect there to be anything surprising there, but it would be good to go over.